proseco/guide.py

```diff
@@ -153,13 +159,58 @@ def run_search_stages(self):
                 self.log(f'Quitting after stage {idx - 1} with {already_selected} stars', level=1)
                 break
         self.log('Done with search stages')
-        selected = cand_guides[cand_guides['stage'] != -1]
-        selected.sort(['stage', 'mag'])
-        if len(selected) >= self.n_guide:
-            return selected[0:n_guide]
-        if len(selected) < self.n_guide:
+        stage_cands = cand_guides[cand_guides['stage'] != -1]
+        stage_cands.sort(['stage', 'mag'])
+        guides = self.select_catalog(stage_cands[0:n_guide + GUIDE.surplus_stars])
+        if len(guides) < self.n_guide:
             self.log(f'Could not find {self.n_guide} candidates after all search stages')
-            return selected
+        return guides
+
+    def select_catalog(self, stage_cands):
+        """
+        For the candidates selected at any stage (stage-assigned candidates?) select the
+        first combination that satisfies the most number of extra tests.
+        Here the extra tests are just the 3 checks for guide stars that are
+        too tightly clustered.
+        """
+        self.log(f'Selecting catalog from {len(stage_cands)} stage-selected stars')
+
+        def index_combinations(n, m):
+            if n == m:
+                yield range(n)
```

Comment:
Sacrilege?

Reply:
Yes.  That was my bug, it should be `range(m, n + 1)` not `range(m + , n + 1)`.

Reply:
Ah.  That makes more sense.  Thanks!